build/all/terraform-bundle.hcl

```diff
@@ -14,10 +14,10 @@ providers {
     versions = ["2.68.0"]
   }
   google = {
-    versions = ["3.62.0"]
+    versions = ["4.19.0"]
```

Comment:
Have you tested if moving to a new major version causing trouble?
Any migrations required in https://github.com/gardener/gardener-extension-provider-gcp/blob/master/pkg/internal/infrastructure/templates/main.tpl.tf ?

Reply:
Hi I did not test yet, I noticed there is a e2e test for AWS only. 
What do you suggest for testing ? I can look at the breaking changes upgrading to 4.x.x but I have the feeling there might be a better way.
Also do you suggest upgrading all the way to 4.26.0 (latest) instead ?

Reply:
I went through the template, and assessed against the breaking changes. I do not foresee issues with the current set of resources in use.

Reply:
Well I guess it would make sense to create a GCP `Infrastructure` with the old tf gcp provider version and then upgrade it with the new one.

Reply:
I created an gcp `Infrastructure` with a terraformer that used gcp-tf-provider `3.62.0` and upgraded it by using a terraformer with gcp-tf-provider `4.19.0`. Seems to work as expected.
Also creation and deletion of a new/existing `Infrastructure` seems to work with `4.19.0`.
Then I guess we can move on.